flocker/node/agents/ebs.py

```diff
@@ -225,7 +225,19 @@ def _wait_for_volume(volume,
     # start_status -> transient_status -> end_status.
     start_time = time.time()
     while time.time() - start_time < VOLUME_STATE_CHANGE_TIMEOUT:
-        volume.update()
+        try:
+            volume.update()
+        except EC2ResponseError as e:
+            # If the volume is being deleted, upon successful deletion,
+            # we will fail to see the volume, as indicated by
+            # ``EC2ResponseError`` code ``InvalidVolume.NotFound``.
+            # (http://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html
+            # for error details).
+            if (transient_status == u'deleting' and
```

Comment:
Having this special case deleting seems... unnecessary? How about just raising `UnknownVolume` if you ever get that code, and then you can just do `try: _wait_for(...) except UnknownVolume: return` in `destroy_volume()`.